lib/commands/kendoui-base.ts

```diff
@@ -0,0 +1,57 @@
+///<reference path="../.d.ts"/>
+"use strict";
+import Future = require("fibers/future");
+import assert = require("assert");
+
+export class KendoUIBaseCommand implements ICommand {
+
+	constructor(protected $errors: IErrors,
+		protected $project: Project.IProject,
+		protected $kendoUIService: IKendoUIService,
+		protected $loginManager: ILoginManager,
+		protected $options: IOptions) {	}
+
+	allowedParameters: ICommandParameter[] = [];
+
+	canExecute(args: string[]): IFuture<boolean> {
+		return ((): boolean => {
+			if(args && args.length > 0) {
+				this.$errors.fail("This command does not accept parameters.");
+			}
+
+			if(this.$options.core && this.$options.professional) {
+				this.$errors.fail("You cannot specify core and professional options simultaneously.");
+			}
+
+			return true;
+		}).future<boolean>()();
+	}
+
+	execute(args: string[]): IFuture<void> {
+		assert.fail("","", "You should never get here. Please contact Telerik support and send the output of your command, executed with `--log trace`.");
+		return Future.fromResult();
+	}
+
+	public getKendoPackages(): IFuture<Server.IKendoDownloadablePackageData[]>{
+		return (() => {
+			this.$loginManager.ensureLoggedIn().wait();
+			this.$project.ensureProject();
+			if (!this.$project.capabilities.updateKendo) {
+				this.$errors.fail("This operation is applicable only to hybrid projects.");
```

Comment:
is it applicable to mobile website projects ? Maybe you can use `this.$project.projectData.Framework` in the message.